src/main/java/com/robertx22/age_of_exile/gui/texts/StatCategory.java

```diff
@@ -16,32 +23,56 @@
 import com.robertx22.age_of_exile.database.data.stats.types.resources.mana.ManaRegen;
 import com.robertx22.age_of_exile.database.registry.ExileDB;
 import com.robertx22.age_of_exile.saveclasses.item_classes.tooltips.TooltipStatInfo;
+import com.robertx22.library_of_exile.registry.IGUID;
 import lombok.AllArgsConstructor;
 
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.Map;
+import java.lang.reflect.ParameterizedType;
+import java.util.*;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 //use to sort the stat when showing MergeStats.
 @AllArgsConstructor
 public enum StatCategory {
-    SDI(x -> Sets.newHashSet("all_attributes", "strength", "dexterity", "intelligence").contains(x)),
-    CORE(x -> Sets.newHashSet(WeaponDamage.GUID, Health.GUID, MagicShield.GUID, Mana.GUID, Energy.GUID, Armor.GUID, DodgeRating.GUID, "critical_hit", "critical_damage").contains(x)),
-    REGEN(x -> Sets.newHashSet(HealthRegen.getInstance().GUID(), MagicShieldRegen.getInstance().GUID(), ManaRegen.getInstance().GUID(), EnergyRegen.getInstance().GUID(), RegeneratePercentStat.HEALTH.GUID(), RegeneratePercentStat.MAGIC_SHIELD.GUID(), RegeneratePercentStat.MANA.GUID(), RegeneratePercentStat.ENERGY.GUID()).contains(x)),
-    RESISTS(x -> x.contains("_resist")),
+    SDI(x -> AllAttributes.getInstance().equals(x) || new HashSet<>(AllAttributes.getInstance().coreStatsThatBenefit()).contains(x)),
+    CORE(x -> Sets.newHashSet(WeaponDamage.getInstance(), Health.getInstance(), MagicShield.getInstance(), Mana.getInstance(), Energy.getInstance(), Armor.getInstance(), DodgeRating.getInstance(), OffenseStats.CRIT_CHANCE.get(), OffenseStats.CRIT_DAMAGE.get()).contains(x)),
+    REGEN(x -> Sets.newHashSet(HealthRegen.getInstance(), MagicShieldRegen.getInstance(), ManaRegen.getInstance(), EnergyRegen.getInstance(), RegeneratePercentStat.HEALTH, RegeneratePercentStat.MAGIC_SHIELD, RegeneratePercentStat.MANA, RegeneratePercentStat.ENERGY).contains(x)),
+    DEFENSE(x -> {
+        boolean b = x instanceof ElementalResist;
+        DefenseStats defenseStats = new DefenseStats();
+        Set<Stat> dataPackStatAccessors = Arrays.stream(DefenseStats.class.getFields())
```

Comment:
I don't like this reflection here at all.

Better just add a new field to stat called tooltip order. You don't have to implement all of them

Reply:
Could I use StatGuiGroupSection?

Reply:
Ok, but i'll probably quickly delete StatGuiGroupSection and make it unsorted or do it my way myself later. Should take only an hour anyway

Reply:
You can use StatGuiGroupSection , but don't edit it. I don't want it to have reflection like what you did there

Reply:
ok